src/main/java/org/openstreetmap/atlas/checks/validation/tag/BridgeDetailedInfoCheck.java

```diff
@@ -0,0 +1,108 @@
+            HighwayTag.TRUNK, HighwayTag.PRIMARY, HighwayTag.SECONDARY);
+    private static final Double MINIMUM_LENGTH = 500.0;
+    public static final String BRIDGE_STRUCTURE_TAG = "bridge:structure";
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(
+            "The length of this bridge (OSM ID: {0,number,#}) makes it deserve more details than just 'bridge=yes'. Add an appropriate 'bridge=*' or 'bridge:structure=*' tag.");
+    private final Distance minimumLength;
+
+    /**
+     * The default constructor that must be supplied. The Atlas Checks framework will generate the
+     * checks with this constructor, supplying a configuration that can be used to adjust any
+     * parameters that the check uses during operation.
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public BridgeDetailedInfoCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.minimumLength = configurationValue(configuration, "bridge.length.minimum.meters",
+                MINIMUM_LENGTH, Distance::meters);
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        // only master edges shall be flagged to avoid duplicate flags on the same OSM Way
+        return object instanceof Edge && ((Edge) object).isMasterEdge() && isGenericBridge(object)
+                && (isRailway(object) || isMajorHighway(object));
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Optional<String> bridgeStructureTag = object.getTag(BRIDGE_STRUCTURE_TAG);
+        if (((Edge) object).length().isGreaterThan(this.minimumLength)
+                && !bridgeStructureTag.isPresent())
+        {
+            return Optional
+                    .of(createFlag(object, getLocalizedInstruction(0, object.getOsmIdentifier())));
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    private boolean isGenericBridge(final AtlasObject object)
+    {
+        return Validators.isOfType(object, BridgeTag.class, BridgeTag.YES);
+    }
+
+    private boolean isMajorHighway(final AtlasObject object)
+    {
+        return MAJOR_HIGHWAYS.contains(((Edge) object).highwayTag());
```

Comment:
One other way to check this would be to use Validators class. Predicate can be declared as a constant at the top like:
`private static final Predicate<AtlasObject> MAJOR_HIGHWAYS = object -> Validators.isOfType(
            object, Highway.class, HighwayTag.MOTORWAY,
            HighwayTag.TRUNK, HighwayTag.PRIMARY, HighwayTag.SECONDARY)`